tests/st/bgp/test_node_status_resilience.py

```diff
@@ -0,0 +1,123 @@
+                DockerHost('host2',
+                           additional_docker_options=CLUSTER_STORE_DOCKER_OPTIONS,
+                           start_calico=False) as host2, \
+                DockerHost('host3',
+                           additional_docker_options=CLUSTER_STORE_DOCKER_OPTIONS,
+                           start_calico=True) as host3:
+
+            # Set the default AS number.
+            host1.calicoctl("config set asNumber %s" % LARGE_AS_NUM)
+
+            # Start host1 using the inherited AS, and host2 using a specified
+            # AS (same as default).  These hosts use the gobgp backend, whereas
+            # host3 uses BIRD.
+            host1.start_calico_node("--backend=gobgp")
+            host2.start_calico_node("--backend=gobgp --as=%s" % LARGE_AS_NUM)
+
+            # Create a network and a couple of workloads on each host.
+            network1 = host1.create_network("subnet1")
+            workload_host1 = host1.create_workload("workload1", network=network1)
+            workload_host2 = host2.create_workload("workload2", network=network1)
+            workload_host3 = host3.create_workload("workload3", network=network1)
+
+            # Allow network to converge
+            self.assert_true(workload_host1.check_can_ping(workload_host2.ip, retries=10))
+
+            # Check connectivity in both directions
+            self.assert_ip_connectivity(workload_list=[workload_host1,
+                                                       workload_host2,
+                                                       workload_host3],
+                                        ip_pass_list=[workload_host1.ip,
+                                                      workload_host2.ip,
+                                                      workload_host3.ip])
+
+            # Check the BGP status on the BIRD/GoBGP host.
+            _log.debug("==== docker exec -it calico-node ps -a  ====")
+            host3.execute("docker exec -it calico-node ps -a")
+
+            hosts = [host1, host2, host3]
+            workloads = [workload_host1, workload_host2, workload_host3]
+
+            def check_connected():
+                for target in hosts:
+                    expected = [("node-to-node mesh", h.ip, "Established") for h in hosts if h is not target]
+                    _log.debug("expected : %s", expected)
+                    check_bird_status(target, expected)
+
+            def delete_workload(host, host_workload):
+                host.calicoctl("ipam release --ip=%s" % host_workload.ip)
+                host.execute("docker rm -f %s" % host_workload.name)
+                host.workloads.remove(host_workload)
+
+            for iteration in range(1, 4):
+                _log.debug("Iteration %s", iteration)
+                _log.debug("identify and pkill bird pid")
+                host3.execute("docker exec -it calico-node pgrep bird")
+                host3.execute("docker exec -it calico-node pkill bird")
+
```

Comment:
I'd kinda like to see a negative test in here, but I'm not totally sure if its possible.  What if the `pkill` fails to actually kill BIRD?  That might give us a false positive.  If we can check that the connection goes down momentarily, we could be fairly sure that BIRD has died.